src/libsystemd-network/network-internal.c

```diff
@@ -272,25 +273,35 @@ int config_parse_hwaddr(const char *unit,
                         void *userdata) {
         struct ether_addr **hwaddr = data;
         struct ether_addr *n;
-        int r;
+        int r, before = 0, after = 0;
+        const char *start = rvalue, *end;
 
         assert(filename);
         assert(lvalue);
         assert(rvalue);
         assert(data);
 
+        while (isspace(*start))
+                start++;
+
         n = new0(struct ether_addr, 1);
         if (!n)
                 return log_oom();
 
-        r = sscanf(rvalue, "%02hhx:%02hhx:%02hhx:%02hhx:%02hhx:%02hhx",
+        r = sscanf(start, "%n%02hhx:%02hhx:%02hhx:%02hhx:%02hhx:%02hhx%n",
+                   &before,
                    &n->ether_addr_octet[0],
                    &n->ether_addr_octet[1],
                    &n->ether_addr_octet[2],
                    &n->ether_addr_octet[3],
                    &n->ether_addr_octet[4],
-                   &n->ether_addr_octet[5]);
-        if (r != 6) {
+                   &n->ether_addr_octet[5],
+                   &after);
+        end = start + after;
```

Comment:
Hmm, I think we should probably replace sscanf() here with ether_aton_r(), unless there's a good reason why that parser isn't good enough for us. @teg, any reason why you used sscanf() here?